lib/nrf_modem_lib/nrf_modem_os.c

```diff
@@ -571,20 +571,20 @@ void nrf_modem_os_init(void)
 
 #if defined(CONFIG_NRF_MODEM_LIB_SHM_TX_DUMP_PERIODIC) || \
 	defined(CONFIG_NRF_MODEM_LIB_HEAP_DUMP_PERIODIC)
-	k_work_q_start(&modem_diag_worqk, work_q_stack_area,
-		       K_THREAD_STACK_SIZEOF(work_q_stack_area),
-		       K_LOWEST_APPLICATION_THREAD_PRIO);
+	k_work_queue_start(&modem_diag_worqk, work_q_stack_area,
+			   K_THREAD_STACK_SIZEOF(work_q_stack_area),
+			   K_LOWEST_APPLICATION_THREAD_PRIO, NULL);
 #endif
 
 #ifdef CONFIG_NRF_MODEM_LIB_SHM_TX_DUMP_PERIODIC
-	k_delayed_work_init(&shmem_task.work, diag_task);
-	k_delayed_work_submit(&shmem_task.work,
+	k_work_init_delayable(&shmem_task.work, diag_task);
+	k_work_reschedule(&shmem_task.work,
 		K_MSEC(CONFIG_NRF_MODEM_LIB_SHMEM_TX_DUMP_PERIOD_MS));
 #endif
 
 #ifdef CONFIG_NRF_MODEM_LIB_HEAP_DUMP_PERIODIC
-	k_delayed_work_init(&heap_task.work, diag_task);
-	k_delayed_work_submit(&heap_task.work,
+	k_work_init_delayable(&heap_task.work, diag_task);
+	k_work_reschedule(&heap_task.work,
 		K_MSEC(CONFIG_NRF_MODEM_LIB_HEAP_DUMP_PERIOD_MS));
```

Comment:
Both these init+reschedule ones could be init+schedule, but that's not a blocker.